### PR TITLE
Remove unused parameter in analyzer loading code

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
IDE always passes 'true' for shadow-copy, so i'm just removing that parameter from the method signature.  This also conceptually makes later changed i'm making simpler as we always know the data is shadow copied.

### CODE REVIEW

**Review by dibarbet — APPROVED** [review]

File: src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerWorkspaceFactory.cs [hunk 1]
```diff
@@ -56,7 +55,7 @@ internal sealed class LanguageServerWorkspaceFactory
     public async Task InitializeSolutionLevelAnalyzersAsync(ImmutableArray<string> analyzerPaths)
     {
         var references = new List<AnalyzerFileReference>();
-        var analyzerLoader = Workspace.Services.GetRequiredService<IAnalyzerAssemblyLoaderProvider>().GetLoader(shadowCopy: true);
+        var analyzerLoader = Workspace.Services.GetRequiredService<IAnalyzerAssemblyLoaderProvider>().GetShadowCopyLoader();
```
Comment (CyrusNajmabadi) [inline src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerWorkspaceFactory.cs:58]: also renamed the method to make it clear you're always gettint he loader with shadow-copying semantics.

**Comment by CyrusNajmabadi** [comment]
@ToddGrun  @jasonmalinowski @tmat ptal.  Thanks @dibarbet

File: src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerAssemblyLoaderService.cs [hunk 2]
```diff
@@ -3,39 +3,28 @@
...
+/// </summary>
+[ExportWorkspaceService(typeof(IAnalyzerAssemblyLoaderProvider), [WorkspaceKind.RemoteWorkspace]), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class RemoteAnalyzerAssemblyLoaderService(
+    [ImportMany] IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
+    : IAnalyzerAssemblyLoaderProvider
+{
+    private readonly ShadowCopyAnalyzerAssemblyLoader _shadowCopyLoader =
+        new(Path.Combine(Path.GetTempPath(), "VS", "AnalyzerAssemblyLoader"), externalResolvers.ToImmutableArray());
```
Comment (jaredpar) [inline src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerAssemblyLoaderService.cs:26]: Do you use this even on Linux? On Linux we don't even need to shadow copy, we can just load from stream. 

Mostly just curious about this question.

Comment (CyrusNajmabadi) [reply src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerAssemblyLoaderService.cs:26]: on linux we have no OOP.  it's strictly a windows thing due to VS being .net fx, and wanting a real .net core process to run in :)

**Pull request merged**